### PR TITLE
Helpers 2.1 (v30.0.3)

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

File: scripts/backup [hunk 1]
```diff
@@ -4,72 +4,76 @@ source ../settings/scripts/_common.sh
...
+ynh_backup "/etc/nginx/conf.d/$domain.d/$app.d"
 
 #=================================================
 # BACKUP THE NGINX CONFIGURATION
 #=================================================
 
-ynh_backup --src_path="/etc/nginx/conf.d/$domain.d/$app.conf"
-ynh_backup --src_path="/etc/nginx/conf.d/$domain.d/$app.d"
+ynh_backup "/etc/nginx/conf.d/$domain.d/$app.conf"
+ynh_backup "/etc/nginx/conf.d/$domain.d/$app.d"
```
Comment (alexAubin) [inline scripts/backup]: duplicated ? :thinking:

File: scripts/backup [hunk 2]
```diff
@@ -4,72 +4,76 @@ source ../settings/scripts/_common.sh
...
 # SYSTEM CONFIGURATION
 #=================================================
 
-ynh_backup --src_path="/etc/php/$phpversion/fpm/pool.d/$app.conf"
+ynh_backup "/etc/php/$php_version/fpm/pool.d/$app.conf"
 
-ynh_backup --src_path="/etc/nginx/conf.d/$domain.d/$app.conf"
-ynh_backup --src_path="/etc/nginx/conf.d/$domain.d/$app.d"
+ynh_backup "/etc/nginx/conf.d/$domain.d/$app.conf"
+ynh_backup "/etc/nginx/conf.d/$domain.d/$app.d"
```
Comment (alexAubin) [inline scripts/backup:28]: uuuh again ? :grimacing:

File: scripts/change_url [hunk 3]
```diff
@@ -61,49 +61,49 @@ then
...
+    ynh_systemctl --service="${app}-notify-push" --action=restart --wait_until="Push daemon for Nextcloud clients." --log_path="systemd" --action=restart
 
     exec_occ config:app:set notify_push base_endpoint --value https://$domain${path_url%/}/push
 
     if ! exec_occ notify_push:self-test; then
-        ynh_print_warn --message="The High Performance Backend service is still not working properly. Please log in with a user to your NextCloud instance, restart the High Performance Backend service with \"systemctl restart ${app}-notify-push.service\", and run \"sudo -u $app php${phpversion} $install_dir/occ notify_push:self-test\" to verify that everything is green."
+        ynh_print_warn "The High Performance Backend service is still not working properly. Please log in with a user to your NextCloud instance, restart the High Performance Backend service with \"systemctl restart ${app}-notify-push.service\", and run \"ynh_exec_as_app php${php_version} $install_dir/occ notify_push:self-test\" to verify that everything is green."
```
Comment (alexAubin) [inline scripts/change_url:91]:
```suggestion
        ynh_print_warn "The High Performance Backend service is still not working properly. Please log in with a user to your NextCloud instance, restart the High Performance Backend service with \"systemctl restart ${app}-notify-push.service\", and run \"sudo -u $app php${php_version} $install_dir/occ notify_push:self-test\" to verify that everything is green."
```

File: scripts/config [hunk 4]
```diff
@@ -142,27 +136,27 @@ set__enable_notify_push() {
...
-        ynh_systemd_action --service_name="${app}-notify-push" --action=restart --line_match="Push daemon for Nextcloud clients." --log_path="systemd"
-        
+        ynh_systemctl --service="${app}-notify-push-watcher" --action=restart
+        ynh_systemctl --service="${app}-notify-push" --action=restart --wait_until="Push daemon for Nextcloud clients." --log_path="systemd"
+
         if ! exec_occ notify_push:self-test; then
-            ynh_print_warn --message="The High Performance Backend service is still not working properly. Please log in with a user to your NextCloud instance, restart the High Performance Backend service with \"systemctl restart ${app}-notify-push.service\", and run \"sudo -u $app php${phpversion} $install_dir/occ notify_push:self-test\" to verify that everything is green."
+            ynh_print_warn "The High Performance Backend service is still not working properly. Please log in with a user to your NextCloud instance, restart the High Performance Backend service with \"systemctl restart ${app}-notify-push.service\", and run \"ynh_exec_as_app php${php_version} $install_dir/occ notify_push:self-test\" to verify that everything is green."
```
Comment (alexAubin) [inline scripts/config:130]:
```suggestion
            ynh_print_warn "The High Performance Backend service is still not working properly. Please log in with a user to your NextCloud instance, restart the High Performance Backend service with \"systemctl restart ${app}-notify-push.service\", and run \"sudo -u $app php${php_version} $install_dir/occ notify_push:self-test\" to verify that everything is green."
```

File: scripts/restore [hunk 5]
```diff
@@ -48,30 +48,30 @@ fi
...
 
 #=================================================
 # RESTORE THE DATA DIRECTORY
 #=================================================
-ynh_script_progression --message="Restoring data directory..." --weight=2
+ynh_script_progression "Restoring data directory..."
 
-# Use --not_mandatory for the data directory, because if the backup has been made with BACKUP_CORE_ONLY, there's no data into the backup.
-ynh_restore_file --origin_path="$data_dir" --not_mandatory
+# Use || true for the data directory, because if the backup has been made with BACKUP_CORE_ONLY, there's no data into the backup.
```
Comment (alexAubin) [inline scripts/restore:73]:
```suggestion
```

(no need to have || true if i remember correctly, everything is handled implicitly by the helpers)

**Review by alexAubin — APPROVED** [review]

**Comment by ericgaspar** [comment]
!testme

File: manifest.toml [hunk 6]
```diff
@@ -19,7 +19,8 @@ code = "https://github.com/nextcloud/server"
 cpe = "cpe:2.3:a:nextcloud:nextcloud"
 
 [integration]
-yunohost = ">= 11.2.30"
+yunohost = ">= 12.0.7"
```
Comment (alexAubin) [inline manifest.toml:22]: Might need ugh 12.0.8 ? (for the mysql helper stuff which is ... not released yet)

**Comment by ericgaspar** [comment]
@alexAubin mais ou sont les parches?

**Comment by alexAubin** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

File: scripts/upgrade [hunk 7]
```diff
@@ -120,37 +122,35 @@ function list_installed_apps_not_compatible_with_future_version()
     comm -23 <(comm -23 $installed_apps $core_apps_in_current_version) $nextcloud_destination_appcatalog
 }
 
-
 current_version=$(grep OC_VersionString "$install_dir/version.php" | cut -d\' -f2)
 current_major_version=${current_version%%.*}
 
-last_version=$(ynh_read_manifest --manifest_key="resources.sources.main.url" | grep -o '[0-9][0-9]\.[0-9]\.[0-9]+')
+last_version=$(ynh_read_manifest "resources.sources.main.url" | grep -o '[0-9][0-9]\.[0-9]\.[0-9]+')
```
Comment (alexAubin) [inline scripts/upgrade:128]:
```suggestion
last_version=$(ynh_read_manifest "resources.sources.main.url" | grep -Eo '[0-9][0-9]\.[0-9]\.[0-9]+')
```

**Pull request merged**